spec/02-integration/09-hybrid_mode/11-status_spec.lua

```diff
@@ -0,0 +1,146 @@
+        cluster_listen = "127.0.0.1:9005",
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+
+        status_listen = "127.0.0.1:" .. cp_status_port
+      })
+    end
+
+    describe("dp should returns 503 without cp", function()
+
+      lazy_setup(function()
+        assert(start_kong_dp())
+      end)
+
+      lazy_teardown(function()
+          assert(helpers.stop_kong("serve_dp"))
+      end)
+
+      -- now dp should be not ready
+      it("should return 503 on data plane", function()
+        helpers.wait_until(function()
+          local http_client = helpers.http_client('127.0.0.1', dp_status_port)
+
+          local res = http_client:send({
+            method = "GET",
+            path = "/status/ready",
+          })
+
+          local status = res and res.status
+          http_client:close()
+          if status == 503 then
+            return true
+          end
+        end, 10)
+      end)
+    end)
+
+    describe("dp status ready endpoint for no config", function()
+
+      lazy_setup(function()
+        assert(start_kong_dp())
+        assert(start_kong_cp())
+      end)
+  
+      lazy_teardown(function()
+          assert(helpers.stop_kong("serve_cp"))
+          assert(helpers.stop_kong("serve_dp"))
+      end)
+
+      -- now cp should be ready
+
+      it("returns 200 on control plane", function()
+        helpers.wait_until(function()
+          local http_client = helpers.http_client('127.0.0.1', cp_status_port)
+
+          local res = http_client:send({
+            method = "GET",
+            path = "/status/ready",
+          })
+
+          local status = res and res.status
+          http_client:close()
+          if status == 200 then
+            return true
+          end
+        end, 10)
+      end)
+
+      -- now dp receive config from cp, so dp should be ready
+
+      it("should return 200 on data plane after configuring", function()
+        helpers.wait_until(function()
+          local http_client = helpers.http_client('127.0.0.1', dp_status_port)
+
+          local res = http_client:send({
+            method = "GET",
+            path = "/status/ready",
+          })
+
+          local status = res and res.status
+          http_client:close()
+          if status == 200 then
+            return true
+          end
+        end, 10)
+
+        assert(helpers.stop_kong("serve_cp", nil, nil, "QUIT", false))
+
+        -- DP should keep return 200 after CP is shut down
+        helpers.wait_until(function()
+
+          local http_client = helpers.http_client('127.0.0.1', dp_status_port)
+
+          local res = http_client:send({
+            method = "GET",
+            path = "/status/ready",
+          })
+
+          local status = res and res.status
+          http_client:close()
+          if status == 200 then
+            return true
+          end
+        end, 10)
+
+        -- recovery state between tests
+        assert(start_kong_cp())
+        ngx.sleep(5)
```

Comment:
Here is another `sleep`, do we have the other choice for it?

Reply:
+1